### ISSUE TITLE
The Domain Model is coupled with the JPA technique.

### ISSUE THREAD

**Issue by c5ms** [citerus/dddsample-core#148]
I found the domain model is marked with JPA annotations, but the repository is not directly implemented by JPA.
So I think the reason the repository is a plain java interface is that we can choose the repository technique by the application, we say the application provides the actually  runtime and the underlaying implementation, it can be see as an runtime.
But if the repository is not implemented inner the domain layer, the domain entity should not coupled with any concrete technique

```java
public class Cargo implements Entity<Cargo> {

  public long id;

  public String trackingId;
}

public interface CargoRepository {
  void store(Cargo cargo);
}


public class JPACargoRepository  implement CargoRepository{

  public void  store(Cargo cargo){
       JPACargo cargo = CargoConvertor.toJPACargo(cargo);
       this.emtityManager.persist(cargo);
   }

}

```

**Comment by orende** [comment]
Indeed, me and @mackapappa have discussed this for quite some time. The only reason the JPA annotations have been placed in the domain objects is because we wanted to move away from using XML bindings for Hibernate, and this was the recommended way to migrate. However, just like you say, this causes the domain objects to be coupled to the repository implementation, which is not ideal.

We do actually have two branches that demonstrate two alternative ways of implementing the repository layer without coupling the domain objects and repository implementation:

- [replace-hibernate-with-jdbi](https://github.com/citerus/dddsample-core/tree/replace-hibernate-with-jdbi)
- [replace-hibernate-with-jpa](https://github.com/citerus/dddsample-core/tree/replace-hibernate-with-jpa)
